src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java

```diff
@@ -35,16 +34,18 @@ public class LoginController {
     public static final String REDIRECT_URL_COOKIE = "redirectUrl";
     public static final String USER_SERVICE_COOKIE_NAME = "user-service-token";
 
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
     @GetMapping("/login")
     public RedirectView login(final @CookieValue(name = USER_SERVICE_COOKIE_NAME, required = false) String jwt,
-            final @RequestParam Optional<String> redirectUrl,
-            final HttpServletResponse response) {
+                              final @RequestParam Optional<String> redirectUrl,
+                              final HttpServletResponse response) {
         final boolean isTokenValid = jwt != null && customJwtService.isTokenValid(jwt);
         if (!isTokenValid) {
-            final Cookie redirectUrlCookie = new Cookie(REDIRECT_URL_COOKIE,
-                    redirectUrl.orElse(configProperties.getDefaultRedirectUrl()));
-            redirectUrlCookie.setSecure(true);
-            redirectUrlCookie.setHttpOnly(true);
+            final Cookie redirectUrlCookie = WebUtil.buildCookie(
```

Comment:
Just a thought but could make this util function follow the builder pattern... Would be a bit more readable, but at this point I ain't fussed